src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/resources/stylesheet.css

```diff
@@ -425,77 +402,89 @@ div.table-tabs > button.table-tab {
    background: #4D7A97;
    color: #FFFFFF;
 }
-
-.row-color th,
-.alt-color th {
-    font-weight:normal;
-}
-.overview-summary td, .member-summary td, .type-summary td,
-.use-summary td, .constants-summary td, .deprecated-summary td,
-.requires-summary td, .packages-summary td, .provides-summary td,
-.uses-summary td, .system-properties-summary td {
+.two-column-summary {
+    display: grid;
+    grid-template-columns: minmax(15%, max-content) minmax(15%, auto);
+}
+.three-column-summary {
+    display: grid;
+    grid-template-columns: minmax(10%, max-content) minmax(15%, max-content) minmax(15%, auto);
+}
+.four-column-summary {
+    display: grid;
+    grid-template-columns: minmax(10%, max-content) minmax(10%, max-content) minmax(10%, max-content) minmax(10%, auto);
+}
+@media screen and (max-width: 600px) {
+    .two-column-summary {
+        display: grid;
+        grid-template-columns: 1fr;
+    }
+}
+@media screen and (max-width: 800px) {
+    .three-column-summary {
+        display: grid;
+        grid-template-columns: minmax(10%, max-content) minmax(25%, auto);
+    }
+    .three-column-summary .col-last {
+        grid-column-end: span 2;
+    }
+}
+@media screen and (max-width: 1000px) {
+    .four-column-summary {
+        display: grid;
+        grid-template-columns: minmax(15%, max-content) minmax(15%, auto);
+    }
+}
+.summary-table > div {
     text-align:left;
-    padding:0px 0px 12px 10px;
+    padding: 8px 3px 3px 7px;
 }
-th.col-first, th.col-second, th.col-last, th.col-constructor-name, th.col-deprecated-item-name, .use-summary th,
-.constants-summary th, .packages-summary th, td.col-first, td.col-second, td.col-last, .use-summary td,
-.constants-summary td, .system-properties-summary th {
+.col-first, .col-second, .col-last, .col-constructor-name, .col-deprecated-item-name {
     vertical-align:top;
-    padding-right:0px;
+    padding-right:0;
     padding-top:8px;
     padding-bottom:3px;
 }
-th.col-first, th.col-second, th.col-last, th.col-constructor-name, th.col-deprecated-item-name, .constants-summary th,
-.packages-summary th {
+.table-header {
     background:#dee3e9;
-    text-align:left;
-    padding:8px 3px 3px 7px;
-}
-td.col-first, th.col-first {
-    font-size:13px;
+    font-weight: bold;
 }
-td.col-second, th.col-second, td.col-last, th.col-constructor-name, th.col-deprecated-item-name, th.col-last {
+.col-first, .col-first {
     font-size:13px;
 }
-.constants-summary th, .packages-summary th {
+.col-second, .col-second, .col-last, .col-constructor-name, .col-deprecated-item-name, .col-last {
     font-size:13px;
 }
-.provides-summary th.col-first, .provides-summary th.col-last, .provides-summary td.col-first,
-.provides-summary td.col-last {
+.provides-summary .col-first, .provides-summary .col-last, .provides-summary .col-first,
+.provides-summary .col-last {
```

Comment:
Hmm, I thought the `*-summary` names were going away?

Reply:
Many of these `*-summary` classes are still used in other parts, e.g. as first argument to `SubHolderWriter.addSummary(...)`. That said, a few of these are indeed unused and can be removed. This will be included in next changeset.